src/JuMP.jl

```diff
@@ -123,50 +174,108 @@ mutable struct Model <: AbstractModel
     ext::Dict{Symbol, Any}
+end
 
-    # Default constructor.
-    function Model(;
-            mode::ModelMode=Automatic,
-            backend=nothing,
-            optimizer=nothing,
-            bridge_constraints=true)
-        model = new()
-        model.variabletolowerbound = Dict{MOIVAR, MOILB}()
-        model.variabletoupperbound = Dict{MOIVAR, MOIUB}()
-        model.variabletofix = Dict{MOIVAR, MOIFIX}()
-        model.variabletointegrality = Dict{MOIVAR, MOIINT}()
-        model.variabletozeroone = Dict{MOIVAR, MOIBIN}()
-        model.customnames = VariableRef[]
-        if backend != nothing
-            # TODO: It would make more sense to not force users to specify
-            # Direct mode if they also provide a backend.
-            @assert mode == Direct
-            @assert optimizer === nothing
-            @assert MOI.isempty(backend)
-            model.moibackend = backend
-        else
-            @assert mode != Direct
-            universal_fallback = MOIU.UniversalFallback(JuMPMOIModel{Float64}())
-            caching_mode = (mode == Automatic) ? MOIU.Automatic : MOIU.Manual
-            caching_opt = MOIU.CachingOptimizer(universal_fallback,
-                                                caching_mode)
-            if bridge_constraints
-                model.moibackend = MOI.Bridges.fullbridgeoptimizer(caching_opt,
-                                                                   Float64)
-            else
-                model.moibackend = caching_opt
-            end
-            if optimizer !== nothing
-                MOIU.resetoptimizer!(model, optimizer)
-            end
-        end
-        model.optimizehook = nothing
-        model.nlpdata = nothing
-        model.objdict = Dict{Symbol, Any}()
-        model.operator_counter = 0
-        model.ext = Dict{Symbol, Any}()
-        return model
+"""
+    Model(factory::Union{Nothing, OptimizerFactory}, moibackend::MOI.ModelLike)
+
+Return a new JuMP model with factory `factory` and MOI backend `moibackend`.
+This constructor is a low-level constructor used by [`Model()`](@ref),
+[`Model(::OptimizerFactory)`](@ref) and [`direct_model`](@ref).
+"""
+function Model(factory::Union{Nothing, OptimizerFactory}, moibackend::MOI.ModelLike)
+    @assert MOI.isempty(moibackend)
+    return Model(Dict{MOIVAR, MOILB}(),
+                 Dict{MOIVAR, MOIUB}(),
+                 Dict{MOIVAR, MOIFIX}(),
+                 Dict{MOIVAR, MOIINT}(),
+                 Dict{MOIVAR, MOIBIN}(),
+                 VariableRef[],
+                 factory,
+                 moibackend,
+                 nothing,
+                 nothing,
+                 Dict{Symbol, Any}(),
+                 0,
+                 Dict{Symbol, Any}())
+end
+
+"""
+    Model(; caching_mode::MOIU.CachingOptimizerMode=MOIU.Automatic,
+            bridge_constraints::Bool=true)
+
+Return a new JuMP model without any optimizer storing the model in a cache.
```

Comment:
"without any optimizer storing the model in a cache" or just "without any optimizer"?